### ISSUE TITLE
Syntax highlighting broke in Atom@1.32.0

### ISSUE THREAD

**Issue by IanCStewart** [atom/atom#18336]
### Prerequisites

- [x] Reproduced the problem in Safe Mode: https://flight-manual.atom.io/hacking-atom/sections/debugging/#using-safe-mode
- [x] Followed all applicable steps in the debugging guide: https://flight-manual.atom.io/hacking-atom/sections/debugging/
- [x] Checked the FAQs on the message board for common solutions: https://discuss.atom.io/c/faq
- [x] Checked that your issue isn't already filed: https://github.com/issues?utf8=✓&q=is%3Aissue+user%3Aatom
- [x] Checked that there is not already an Atom package that provides the described functionality: https://atom.io/packages

### Description
All syntax highlighting seems to be broken in Atom 1.32.0.

<img width="2032" alt="image" src="https://user-images.githubusercontent.com/14125280/47552082-1a59f600-d904-11e8-9f8c-2a23b82e1c7f.png">

The syntax seems to have a totally different color or no color at all _(white)_ after the last Atom update.

### Steps to Reproduce

1. Create a file (in my case a JS file)
2. Write some code
3. Notice syntax is not working in all cases

**Expected behavior:** Correct syntax highlighting

**Actual behavior:** Some peaces of code get the wrong color or no color at al

**Reproduces how often:** All the time

### Versions

**atom --version:**
```bash
Atom    : 1.32.0
Electron: 2.0.9
Chrome  : 61.0.3163.100
Node    : 8.9.3
```

**apm --version:**
```bash
apm  2.1.2
npm  6.2.0
node 8.9.3 x64
atom 1.32.0
python 2.7.10
git 2.19.1
```

### Additional Information

I'm not sure if this is just related to JS syntax highlighting or that it is way bigger but i have found numerous issues about highlighting issues within your issue tracker. They're mostly related to the latest Atom update.

**Comment by nathanlesage** [comment]
I second that — same version, same problems.

**Comment by thomasjo** [comment]
Syntax highlighting is not broken, but we have changed the default parser implementation, which is why some parts of your code will be highlighted differently. See https://github.com/atom/atom/issues/18307 for more details.

You can switch back to the old parser by unticking _Use Tree Sitter Parsers_ in your core settings, or by running `atom.config.set('core.useTreeSitterParsers', false)` from the developer console.

If you are seeing specific keywords or syntax in specific languages not highlighted correctly, please open an issue on the relevant language repository (e.g. [atom/language-javascript](https://github.com/atom/language-javascript)).

**Comment by ayhamk94** [comment]
Another way to apply @thomasjo  suggestion is by editing the `config.cson` file.
`Atom` > `config...` and then under `core` add `useTreeSitterParsers: false` and save.
Then you  get your lovely syntax highlighting back. 
Note: this works for me using a mac I have not tried on a Windows or Linux.

**Comment by qhenkart** [comment]
It would have been amazing to get some form of notification regarding the "not broken" breaking change to all of our stylesheets and grammar packages. For me it happened in the middle of the work day, and after a fresh re-install, rewriting my stylesheet and a few issue reports, I finally found this thread 🤦‍♂️ .  FYI, It completely breaks the core Go language package. I have made an git issue there but anyone using go should revert the new default setting and turn off auto updates to avoid similar issues in the future

**Comment by allisonphillips** [comment]
@thomasjo I'm having entire chunks of my code highlight in the wrong color for the past few weeks. If I go and edit each individual line that's incorrectly highlighted, it will fix the highlighting for that specific line. I think this is a legitimate issue.

Highlighting for this chunk of code broke when I added a comment line to the top of the block:
<img width="176" alt="screen shot 2018-11-08 at 2 55 38 pm" src="https://user-images.githubusercontent.com/8813886/48232477-a3424a00-e366-11e8-98ae-3fd3975f7a55.png">

Highlighting corrected for a single line when I added and deleted a space on that line:
<img width="118" alt="screen shot 2018-11-08 at 2 57 14 pm" src="https://user-images.githubusercontent.com/8813886/48232597-fc11e280-e366-11e8-939e-43b4c798d2f3.png">

I've been experiencing this with JS and Ruby code highlighting. It seems like the parser is not working correctly at random.

**Comment by Arcanemagus** [comment]
@allisonphillips In the next hotfix release (v1.32.2) there have been several further improvements made to those languages in particular. If you are still seeing issues after updating to that version when it comes out please file an issue in the [respective](https://github.com/atom/language-javascript) [repository](https://github.com/atom/language-ruby) for the language you are seeing a problem with, if there isn't an issue already filed of course 😉.

**Comment by allisonphillips** [comment]
@Arcanemagus It seemed like a parser issue, given the timeline, that it's affecting multiple languages, it's resolved by setting useTreeSitterParsers to false, and that other people (including my coworkers) are also having this problem within the same timeline. I'm glad there are improvements being made since it looked like this was dismissed as a non-issue above, and I will wait for the next release and file tickets if it's still broken at that time. Thank you for the update! :)

**Comment by Ben3eeE** [comment]
@allisonphillips We noticed exactly this issue for ERB on master a few days ago and the fix for it will ship in Atom 1.32.2. It's likely that this fix will apply to javascript as well. If you still encounter the issue then please open a new issue.

> this was dismissed as a non-issue above,

This issue and what you report are entirely different issues. This issue is about colors being different permanently, See https://github.com/atom/atom/issues/18352#issuecomment-436829384 for a more in depth explanation of this. Your report is about syntax highlighting breaking.
